test/test_cli.rb

```diff
@@ -253,6 +253,30 @@ def test_state_file_callback_filtering
     assert_empty keys_not_stripped
   end
 
+  def test_log_formatter_default_single
+    cli = Puma::CLI.new [ ]
+    assert_instance_of Puma::Events::DefaultFormatter, cli.launcher.events.formatter
+  end
+
+  def test_log_formatter_default_clustered
+    skip_on :jruby, :windows
+
+    cli = Puma::CLI.new [ "-w 2" ]
+    assert_instance_of Puma::Events::PidFormatter, cli.launcher.events.formatter
+  end
+
+  def test_log_formatter_custom_single
+    cli = Puma::CLI.new [ "--config", "test/config/custom_log_formatter.rb" ]
+    assert_instance_of Puma::Events::CustomFormatter, cli.launcher.events.formatter
+  end
+
+  def test_log_formatter_custom_clustered
+    skip_on :jruby, :windows
```

Comment:
Would prefer @MSP-Greg's approach to the skips here.

Reply:
done :+1: